crates/types/src/identifiers.rs

```diff
@@ -238,45 +250,96 @@ pub trait ResourceId {
     serde_with::SerializeDisplay,
     serde_with::DeserializeFromStr,
 )]
-pub struct InvocationUuid(Ulid);
+pub struct InvocationUuid(u128);
 
 impl InvocationUuid {
     pub const SIZE_IN_BYTES: usize = size_of::<u128>();
 
-    pub fn new() -> Self {
-        Self(Ulid::new())
-    }
+    const HASH_SEPARATOR: u8 = 0x2c;
 
     pub fn from_slice(b: &[u8]) -> Result<Self, IdDecodeError> {
-        let ulid = Ulid::from_bytes(b.try_into().map_err(|_| IdDecodeError::Length)?);
-        debug_assert!(!ulid.is_nil());
-        Ok(Self(ulid))
+        Ok(Self(u128::from_be_bytes(
+            b.try_into().map_err(|_| IdDecodeError::Length)?,
+        )))
```

Comment:
Should we keep the assertion that it's not 0?

Reply:
no strong opinion. will reintroduce it.